erna/scripts/process_fact_mc.py

```diff
@@ -30,8 +53,9 @@ def make_jobs(jar, xml, data_paths, drs_paths,
     for num, (data, drs) in enumerate(zip(data_partitions, drs_partitions)):
         df = pd.DataFrame({'data_path': data, 'drs_path': drs})
         if output_path:
+            # create the filenames for each single local run
             file_name, _ = path.splitext(path.basename(output_path))
-            file_name += "_{}.json".format(num)
+            file_name = createFilenameFromFormat(filename_format, file_name, num)
```

Comment:
Simply use `filename_format.format(name=file_name, number=num)` or similar. Also the old name of your function is still used here (CamelCase vs snake_case)